### DISCUSSION TITLE
CareKit 3.0

### DISCUSSION THREAD

**Discussion by gavirawson-apple** [carekit-apple/CareKit#690] [Announcements]
CareKit 3.0 introduces new tools to integrate care data into your SwiftUI views, including tools that stream care data and that aggregate progress across care tasks. With the help of these tools, let CareKit handle persistence of user data while you focus on building out engaging visual experiences for your users.

We’d love to hear what you think of the new updates. Feel free to leave a general note or idea here, or open up a new issue if you have any specific questions.

Happy coding!

**Comment by ronzilla-apple** [comment]
Release notes here
https://github.com/carekit-apple/CareKit/blob/main/RELEASE-NOTES.md

**Comment by cbaker6** [comment]
Team, I think all of the new features are amazing! I've taken some time to experiment with the latest features along with adding some fixes in #691. I've updated the [ParseCareKit Sample App](https://github.com/netreconlab/CareKitSample-ParseCareKit) to use the latest changes in the linked PR. For those developers who want to try out CareKit 3.0.0 in a SwiftUI based app, you can fork this [branch](https://github.com/netreconlab/CareKitSample-ParseCareKit/tree/main) and do one of the following:

1. If you want the app to synchronize with a local docker server follow [here](https://github.com/netreconlab/CareKitSample-ParseCareKit/tree/newCareKit#docker) (easy, fork and type `docker compose up`). Simply click `Login Anonymously` when presented with the app login to skip signing up
2. If you want something similar to the [What's New in CareKit](https://developer.apple.com/videos/play/wwdc2020/10151/) WWDC20 video in which the iOS app syncs with an Apple Watch (set the flag `isSyncingWithRemote` in [Constants.swift](https://github.com/netreconlab/CareKitSample-ParseCareKit/blob/e1440ad414c3fa86a42cb8a349423ca7ba10d4b0/OCKSample/Constants.swift#L14-L17) to  `isSyncingWithRemote = false`

For those interested in seeing how to use the new `@CareStoreFetchRequest` in SwiftUI, check the following files:
- [ProfileView using OCKPatient on iOS](https://github.com/netreconlab/CareKitSample-ParseCareKit/blob/1d858a5f35c4984c6aa31b1301c3c8d46c40fee9/OCKSample/Main/Profile/ProfileView.swift#L15-L81)
- [CareView using OCKEvents and SwiftUI CareKit Cards on watchOS](https://github.com/netreconlab/CareKitSample-ParseCareKit/blob/1d858a5f35c4984c6aa31b1301c3c8d46c40fee9/OCKWatchSample%20Extension/Main/Care/CareView.swift#L15-L29)

If you are interested in seeing how to use the updated UIKit CareKit cards see: https://github.com/netreconlab/CareKitSample-ParseCareKit/blob/1d858a5f35c4984c6aa31b1301c3c8d46c40fee9/OCKSample/Main/Care/CareViewController.swift#L202-L264

To see how to take advantage of the new streaming CareKitStore, see: https://github.com/netreconlab/CareKitSample-ParseCareKit/pull/116

Note that everything seems to work fine in Xcode 14.2, but in Xcode 14.3 the app may crash if you create a view that contains SwiftUI and UIKit cards. I've noted the issue here: https://github.com/carekit-apple/CareKit/pull/691#discussion_r1162096553.

**Reply by gavirawson-apple** [reply]
Lovely uses of the new APIs @cbaker6! Keep us posted on your amazing work.
